src/Build/Evaluation/Expander.cs

```diff
@@ -4534,6 +4533,24 @@ private static bool TryConvertToInt(object value, out int arg0)
                 return false;
             }
 
+            private static bool TryConvertToLong(object value, out long arg0)
+            {
+                switch (value)
+                {
+                    case double d:
+                        arg0 = Convert.ToInt64(d);
```

Comment:
Added the min/max check to both TryConvertToLong and TryConvertToInt.

TryConvertToLong  is only used by the arithmetic overloads.

TryConvertToInt is used in many places and the change to add the min/max check is a separate commit in case it needs to be rolled back.